src/include/duckdb/function/partition_stats.hpp

```diff
@@ -31,6 +32,11 @@ struct PartitionStatistics {
 	idx_t count;
 	//! Whether or not the count is exact or approximate
 	CountType count_type;
+
+	optional_ptr<RowGroup> row_group;
```

Comment:
This is DuckDB storage specific so that shouldn't be here - `PartitionStatistics` is generic and needs to be emittable by all scans, also e.g. Parquet scans. Maybe this can be some virtual wrapper instead? e.g.:

```cpp
struct PartitionRowGroup {
    virtual ~ PartitionRowGroup() = default;

   virtual unique_ptr<BaseStatistics> GetColumnStatistics(column_t column_id) = 0;
};
```

Then this can be implemented by the `DataTable`:


```cpp

struct DuckDBPartitionRowGroup : public PartitionRowGroup {
   optional_ptr<RowGroup> row_group;

  unique_ptr<BaseStatistics> GetColumnStatistics(column_t column_id) override {
      return row_group->GetStatistics(column_id);
   }
};
```